components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/dao/impl/APIResourceManagementDAOImpl.java

```diff
@@ -488,6 +490,51 @@ public List<ApplicationBasicInfo> getSubscribedApplications(String apiId) {
         return null;
     }
 
+    @Override
+    public List<APIResource> getScopeMetadata(List<String> scopeNames, Integer tenantId)
+            throws APIResourceMgtException {
+
+        String query = SQLConstants.GET_SCOPE_METADATA;
+        String placeholders = String.join(",", Collections.nCopies(scopeNames.size(), "?"));
+        query = query.replace(SQLConstants.SCOPE_LIST_PLACEHOLDER, placeholders);
+        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(false)) {
+            PreparedStatement prepStmt = dbConnection.prepareStatement(query);
+            prepStmt.setInt(1, tenantId);
+            int scopeIndex = 2;
+            for (String scopeName : scopeNames) {
+                prepStmt.setString(scopeIndex, scopeName);
+                scopeIndex++;
+            }
+            try (ResultSet resultSet = prepStmt.executeQuery()) {
+                Map<String, APIResource> apiResources = new HashMap<>();
+                while (resultSet.next()) {
+                    String apiId = resultSet.getString(SQLConstants.API_RESOURCE_ID_COLUMN_NAME);
+                    Scope scope = new Scope.ScopeBuilder()
+                            .name(resultSet.getString(SQLConstants.SCOPE_QUALIFIED_NAME_COLUMN_NAME))
+                            .displayName(resultSet.getString(SQLConstants.SCOPE_DISPLAY_NAME_COLUMN_NAME))
+                            .description(resultSet.getString(SQLConstants.SCOPE_DESCRIPTION_COLUMN_NAME))
+                            .build();
+                    if (apiResources.containsKey(apiId)) {
+                        apiResources.get(apiId).getScopes().add(scope);
+                    } else {
+                        List<Scope> scopes = new ArrayList<>();
+                        scopes.add(scope);
+                        APIResource apiResource = new APIResource.APIResourceBuilder()
+                                .id(apiId)
+                                .name(resultSet.getString(SQLConstants.API_RESOURCE_NAME_COLUMN_NAME))
+                                .scopes(scopes)
+                                .build();
```

Comment:
can't we pass the scope as a list to the builder

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5003/commits/ea4b23a67c37fdf0e9b8c39e8ee2dccd0167fbda